pkg/controller/dnszone/gcpactuator.go

```diff
@@ -85,18 +85,27 @@ func (a *GCPActuator) Create() error {
 
 	logger.Debug("Managed zone successfully created")
 	a.managedZone = managedZone
+	if err := a.ModifyStatus(); err != nil {
+		logger.WithError(err).Error("failed to sync DNSZone status fields")
+		return err
+	}
+
 	return nil
 }
 
 // Delete implements the Delete call of the actuator interface
 func (a *GCPActuator) Delete() error {
```

Comment:
This function is still using `a.managedZone.Name` when calling `DeleteManagedZone`. It should use `*a.dnsZone.Status.GCP.ZoneName` instead. It may be helpful to create a `zoneName` variable early in the function and just use that instead.